core/services/workflows/v2/secrets.go

```diff
@@ -75,12 +79,37 @@ func (s *secretsFetcher) GetSecrets(ctx context.Context, request *sdkpb.GetSecre
 	return resp, err
 }
 
-func (s *secretsFetcher) getSecrets(ctx context.Context, request *sdkpb.GetSecretsRequest) ([]*sdkpb.SecretResponse, error) {
+func (s *secretsFetcher) getSecretsForBatch(ctx context.Context, request *sdkpb.GetSecretsRequest) ([]*sdkpb.SecretResponse, error) {
 	vaultCap, err := s.capRegistry.GetExecutable(ctx, vault.CapabilityID)
 	if err != nil {
 		return nil, errors.New("failed to get vault capability: " + err.Error())
 	}
 
+	localNode, err := s.capRegistry.LocalNode(ctx)
```

Comment:
@prashantkumar1982 The vault DON capability won't be on the workflow DON, so you won't be able to get the DON ID on this method, you'll need to get it by calling vaultCap.Info() and getting the DON ID from there

For tests or completeness you may want to accommodate both cases and check whether vaultCap.Info() is a local capability (`IsLocal`) and use the localNode don ID if so, and the cap don ID if not.

Lmk if that makes sense, happy to explain further

Reply:
Ahh sorry, yes my miss.
I didn't realize localNode DonId is the workflow don, not the vault don.

So I should be able to just use "vaultCapInfo.DON.ID" right?
I don't even need to look at localNode irrespective right?

I'm doing those changes anyways, please review.

Reply:
Looks like you addressed this correctly 👍